cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/AuthorizeAction.java

```diff
@@ -232,8 +233,13 @@ private void doOpenIDCodeFlow() throws Exception {
             // if authenticated (only possible by cookie) skip login form
             // formulate the authenticate redirect response
             
-            // TODO Alinga
-            // Add group check on rp.accessGroup here
+            // perform group check on rp.accessGroup 
+            if (!OIDCUtil.accessAllowed(rp)) {
+                AuthorizeError authError = new AuthorizeError();
+                authError.error = "login failed, client " + clientID + " is not a member of " + rp.getAccessGroup();
```

Comment:
It's actually the user who is not a member of the group, so the message should be "login failed, not a member of group <groupURI>"

Reply:
Right. Thanks.